modem/design.md

```diff
@@ -0,0 +1,8 @@
+Overview:
+
+![Design diagram](diagram.png)
+
+We create a Transceiver object that will manage the serial connection. It will do command detection and hand off to the appropriate Command object to handle the interaction. The Command object will be subclassed for each type of command we will receive and will handle the processing of the command and generating the response.
+
+The Transceiver object will also own the Firebase object and present it to each command during processing. The Firebase object is our link to the firebase backend.
```

Comment:
I like the idea of splitting parsing of the serial protocol from the command execution, it could come handy if we want to add more protocol based on `SPI` or `I2C`. It would be nice if the `Source` for the `Transceiver` was also abstracted maybe using a `Arduino [Stream](https://www.arduino.cc/en/Reference/Stream)` so that we can easily plug the `SerialTransceiver` with a websocket or telnet transport.